src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java

```diff
@@ -0,0 +1,82 @@
+package it.pagopa.transactions.commands.handlers;
+
+import it.pagopa.generated.transactions.server.model.TransactionStatusDto;
+import it.pagopa.transactions.client.NodoPerPM;
+import it.pagopa.transactions.commands.TransactionActivateResultCommand;
+import it.pagopa.transactions.documents.TransactionActivatedData;
+import it.pagopa.transactions.documents.TransactionActivatedEvent;
+import it.pagopa.transactions.domain.TransactionActivated;
+import it.pagopa.transactions.exceptions.AlreadyProcessedException;
+import it.pagopa.transactions.exceptions.TransactionNotFoundException;
+import it.pagopa.transactions.repositories.PaymentRequestInfo;
+import it.pagopa.transactions.repositories.PaymentRequestsInfoRepository;
+import it.pagopa.transactions.repositories.TransactionsEventStoreRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+@Component
+@Slf4j
+public class TransactionActivateResultHandler
+		implements CommandHandler<TransactionActivateResultCommand, Mono<TransactionActivatedEvent>> {
+
+	@Autowired
+	private TransactionsEventStoreRepository<TransactionActivatedData> transactionEventStoreRepository;
+
+	@Autowired private NodoPerPM nodoPerPM;
+
+	@Autowired private PaymentRequestsInfoRepository paymentRequestsInfoRepository;
+
+	@Override
+	public Mono<TransactionActivatedEvent> handle(TransactionActivateResultCommand command) {
+
+		final TransactionActivated transactionInitialized = command.getData().transactionInitialized();
+
+		final String transactionId = command.getData().transactionInitialized().getTransactionId().toString();
+		TransactionActivatedData data = new TransactionActivatedData();
+		data.setAmount(transactionInitialized.getAmount().value());
+		data.setDescription(transactionInitialized.getDescription().value());
+
+		return Mono.just(command)
+				.filterWhen(commandData -> Mono
+						.just(commandData.getData().transactionInitialized().getStatus() == TransactionStatusDto.ACTIVATION_REQUESTED))
+				.switchIfEmpty(Mono.error(new AlreadyProcessedException(command.getRptId())))
+				.flatMap(commandData -> {
+					final String paymentToken = commandData.getData().activationResultData().getPaymentToken();
+
+					return nodoPerPM.chiediInformazioniPagamento(paymentToken)
+							.doOnError(throwable -> {
+								log.error("chiediInformazioniPagamento failed for paymentToken {}", paymentToken);
+								throw new TransactionNotFoundException("chiediInformazioniPagamento failed for paymentToken " +paymentToken);
+							})
```

Comment:
Are you sure this "intercepts" the async call chain? What about this?

```suggestion
							.onErrorMap(throwable -> {
								log.error("chiediInformazioniPagamento failed for paymentToken {}", paymentToken);
								return new TransactionNotFoundException("chiediInformazioniPagamento failed for paymentToken " +paymentToken);
							})
```